File: benchmarks/poly/sortnums/sortnums.rb
def main
  _t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  n = 1000000
  seed = 12345
  arr = Array.new(n)
  n.times do |i|
    seed = (48271 * seed) % 2147483647
    arr[i] = seed % 1000000
  end

  arr.sort!

  checksum = 0
  arr.each do |v|
    checksum = (checksum * 31 + v) % 1000000007
  end

  puts arr[0]
  puts arr[n - 1]
  puts checksum
  $stderr.puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - _t0}"
end

main

File: benchmarks/poly/sortnums/sortnums.js
function main() {
    const _t0 = process.hrtime.bigint();
    const n = 1000000;
    let seed = 12345;
    const arr = new Array(n);
    for (let i = 0; i < n; i++) {
        seed = (48271 * seed) % 2147483647;
        arr[i] = seed % 1000000;
    }

    arr.sort((a, b) => a - b);

    let checksum = 0;
    for (let i = 0; i < n; i++) {
        checksum = (checksum * 31 + arr[i]) % 1000000007;
    }

    console.log(arr[0]);
    console.log(arr[n - 1]);
    console.log(checksum);
    console.error(`bench_ns=${process.hrtime.bigint() - _t0}`);
}

main();
